include/onnxruntime/core/framework/op_kernel_context.h

```diff
@@ -130,6 +130,12 @@ class OpKernelContext {
    */
   Status GetTempSpaceAllocator(AllocatorPtr* output) const ORT_MUST_USE_RESULT;
 
+  /**
+   Return the allocator associated with the CPU EP with memtype of OrtMemTypeDefault.
+   @remarks Use SafeInt when calculating the size of memory to allocate using AllocatorPtr->Alloc.
+   */
+  Status GetCPUAllocator(AllocatorPtr* output) const ORT_MUST_USE_RESULT;
```

Comment:
>AllocatorPtr* [](http://example.com/codeflow?start=25&length=13)

Why pointer to a smart pointer? Can the out parameter be absent? I think it has to be a reference

Reply:
That looked strange to me as well, but currently that is the signature we have for `GetTempSpaceAllocator()` and it feels strange to have multiple variants for methods that return back allocators. So, I kept it consistent for now without opting for a refactor so close to the release. This way, at least the call sites look clean and consistent.